multiqc/modules/homer/tagdirectory.py

```diff
@@ -0,0 +1,568 @@
+                self.add_data_source(f, s_name, section='taginfo')
+                self.tagdir_data['taginfo_total'][s_name] = parsed_data[0]
+                self.tagdir_data['taginfo_total_norm'][s_name] = self.normalize(parsed_data[0])
+                self.tagdir_data['taginfo_uniq'][s_name] = parsed_data[1]
+                self.tagdir_data['taginfo_uniq_norm'][s_name] = self.normalize(parsed_data[1])
+
+
+        for f in self.find_log_files('homer/tagInfo', filehandles=True):
+            s_name = os.path.basename(f['root'])
+            s_name = self.clean_s_name(s_name, f['root'])                
+            ## collected tag_info data for general stats table and store under 'header'
+            parsed_data = self.parse_tag_info(f)
+            if parsed_data is not None:
+                self.tagdir_data['header'][s_name] = parsed_data
+
+
+        self.tagdir_data = self.ignore_samples(self.tagdir_data)
+        description = '<p>This plot shows the distribution of tags along chromosomes.</p>'
+        helptext = '<p>This is a good quality control for tag distribution and could be a good indication of large duplications or deletions.</p>'
+
+        self.add_section (
+        name = 'Tag Info Chromosomal Coverage',
+        anchor = 'homer-tagInfo',
+        description = description,
+        helptext = helptext,
+        plot = self.tag_info_chart()
+        )
+
+
+        # Find and parse homer tag FreqDistribution_1000 reports
+        for f in self.find_log_files('homer/FreqDistribution', filehandles=True):
+            s_name = os.path.basename(f['root'])
+            s_name = self.clean_s_name(s_name, f['root'])
+            parsed_data = self.parse_FreqDist(f)
+            if parsed_data is not None:
+                if s_name in self.tagdir_data['FreqDistribution']:
+                    log.debug("Duplicate Freq Distribution sample log found! Overwriting: {}".format(s_name))
+
+                self.add_data_source(f, s_name, section='FreqDistribution')
+                self.tagdir_data['FreqDistribution'][s_name] = parsed_data
+
+
+        
+        self.tagdir_data = self.ignore_samples(self.tagdir_data)
+        description = '<p>This plot shows the distribution of distance between PE tags.</p>'
+        helptext = '<p>It is expected the the frequency of PE tags decays with increasing distance between the PE tags. This plot gives an idea of the proportion of short-range versus long-range interactions.</p>'
+
+        self.add_section (
+        name = 'Frequency Distribution',
+        anchor = 'homer-FreqDistribution',
+        description = description,
+        helptext = helptext,
+        plot = self.FreqDist_chart()
+        )
+
+
+        # Basic Stats Table
+        self.homer_stats_table_tagInfo()
+
+        if len(self.tagdir_data) > 0:
+
+            # Write parsed report data to a file
+            self.write_data_file(self.tagdir_data, 'multiqc_homer_tagdirectory')
```

Comment:
I don't think that this function will work for your nested data structure. It expects a two-dimensional dictionary with the keys as the sample names.

Usually you need to be fairly selective with this function - eg. only saving single-digit statistics, not continuous plot data.

Reply:
It looks like this function isn't being called in `findpeaks.py` - it would would be great to add this there and save all parsed values to a file!

Reply:
I looked into the written files for the other modules. I can see that they are single values. The Homer values are stored in the general_stat tables files so I am not sure it is worth it to save them again in a separate homer file?